main.cc

```diff
@@ -1323,9 +1323,7 @@ int main(int ac, char** av) {
 
             cql_transport::controller cql_server_ctl(auth_service, mm_notifier, gossiper.local(), qp, service_memory_limiter, sl_controller, lifecycle_notifier, *cfg);
 
-            ss.local().register_client_shutdown_hook("native transport", [&cql_server_ctl] {
-                cql_server_ctl.stop_server().get();
-            });
+            ss.local().register_protocol_server(cql_server_ctl);
 
```

Comment:
Maybe one day as part of dismantling storage_server we'll move these calls to a protocol_server_service.